Small improvements

krober10nd opened the pull request:
Adding some sanity checks.

Commit:
Add warning to CalcCFL

krober10nd commented:
I think it'd be useful to by default plot in the wsg84 if the mesh was read in from a fort.14 otherwise it gets this information from the .mat file. Currently, if the mesh was read in from a fort.14 file and you go to do plot(m) it errors because there's no projection information. IMO, at the minimum this should plot but not in the projection format. 

Also, we should either: not hard code the extension ".14" in the reading and writing operations OR pass another "type" argument to write to enable support for different models (like SCHISM, WW3 etc.).

WPringle commented:
> I think it'd be useful to by default plot in the wsg84 if the mesh was read in from a fort.14 otherwise it gets this information from the .mat file. Currently, if the mesh was read in from a fort.14 file and you go to do plot(m) it errors because there's no projection information. IMO, at the minimum this should plot but not in the projection format.
> 
> Also, we should either: not hard code the extension ".14" in the reading and writing operations OR pass another "type" argument to write to enable support for different models (like SCHISM, WW3 etc.).

I agree, can we add these changes to this pull request?

Commit:
plot in geo. by default

... otherwise use projection info

krober10nd commented:
ok updated the plotting...

Commit:
varargin for msh class constructor

krober10nd commented:
A small but big change but hopefully an improvement...If the user only passes one input to msh, it's assumed it's the filename of the mesh file. Note the mesh is specified WITH the .14 extension (in the case it's an adcirc mesh)

```
m = msh('meshes/GSB_V4_SHToBP_TN_V2.14');
```

note, that this update now opens an easy way to support/input data into the mesh class from other files besides adcirc's fort.14 by reading the suffix of the single input argument. 

If the user wants to read in multiple files at once then they'll have to use name/value arguments...the ``fname`` must be specified like so...

```
m = msh('fname','meshes/GSB_V4_SHToBP_TN_V2.14','type','14');
```

However, this makes the assumption that all files are named (e.g., 13,15,24,etc.) with the same prefix, which I know isn't always the case. So this can still be improved. 

Lastly, now the user can start up a mesh class by doing this from points and elements in their workspace:

```
m = msh('points',p,'elements',t); 
```

which is a lot less awkward than constructing a class loading data into the fields of the class. It also opens the door to some basic checking to make sure the mesh isn't broken. 

The blank class constructor is initialized with no arguments.

WPringle commented:
Regarding the reading in of multiple files; it seems that since you have the filename with '.14' at the end how would it read e.g., the '.13' file since it doesn't appear to extract out only the prefix.

I think it would be better if you could provide the suffix only (to read in multiple files while using the 'type') or the filename with extension (to read in one file, don't need to specify type and it will work it out from extension). 

Or we could just do it that you can provide a cell-list of arbitrary filenames and it would them all in but you would also specify the type of each file using type, that way you can name them whatever you want, even without .14 suffix etc.

krober10nd commented:
I like the third idea there of specifying the files as a cell array if you
want to read in aux adcirc files along with the mesh. This would decouple
the filenames among 14 and 13 files.

On Sun, 12 Jul 2020 at 3:08 PM William Pringle <notifications@github.com>
wrote:

> Regarding the reading in of multiple files; it seems that since you have
> the filename with '.14' at the end how would it read e.g., the '.13' file
> since it doesn't appear to extract out only the prefix.
>
> I think it would be better if you could provide the suffix only (to read
> in multiple files while using the 'type') or the filename with extension
> (to read in one file, don't need to specify type and it will work it out
> from extension).
>
> Or we could just do it that you can provide a cell-list of arbitrary
> filenames and it would them all in but you would also specify the type of
> each file using type, that way you can name them whatever you want, even
> without .14 suffix etc.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CHLNDDEV/OceanMesh2D/pull/93#issuecomment-657256295>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEOBZ7CXBVO47DQ4VWTYGSLR3H32LANCNFSM4OQM3CLQ>
> .
>
-- 
--------------------
Keith Roberts
Post-doctoral researcher
University of São Paulo
krober@usp.br <krober10@nd.edu>

krober10nd commented:
The reason I had done it with specifying the fully file name with the
suffix is that then you could detect which triangulation file format you’re
reading.

On Sun, 12 Jul 2020 at 3:20 PM Keith Roberts <Keith.J.Roberts.133@nd.edu>
wrote:

> I like the third idea there of specifying the files as a cell array if you
> want to read in aux adcirc files along with the mesh. This would decouple
> the filenames among 14 and 13 files.
>
> On Sun, 12 Jul 2020 at 3:08 PM William Pringle <notifications@github.com>
> wrote:
>
>> Regarding the reading in of multiple files; it seems that since you have
>> the filename with '.14' at the end how would it read e.g., the '.13' file
>> since it doesn't appear to extract out only the prefix.
>>
>> I think it would be better if you could provide the suffix only (to read
>> in multiple files while using the 'type') or the filename with extension
>> (to read in one file, don't need to specify type and it will work it out
>> from extension).
>>
>> Or we could just do it that you can provide a cell-list of arbitrary
>> filenames and it would them all in but you would also specify the type of
>> each file using type, that way you can name them whatever you want, even
>> without .14 suffix etc.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/CHLNDDEV/OceanMesh2D/pull/93#issuecomment-657256295>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEOBZ7CXBVO47DQ4VWTYGSLR3H32LANCNFSM4OQM3CLQ>
>> .
>>
> --
> --------------------
> Keith Roberts
> Post-doctoral researcher
> University of São Paulo
> krober@usp.br <krober10@nd.edu>
>
> --
--------------------
Keith Roberts
Post-doctoral researcher
University of São Paulo
krober@usp.br <krober10@nd.edu>

Commit:
support for reading aux files with names decoupled from fort.14

krober10nd commented:
Ok, so now here's the behavior:

1. If one file is passed, this is assumed to be a mesh (e.g., fort.14). The format is determined by the suffix. 
```
m = msh('ECGC.14')
```
No mickey mouse magic behavior or assumptions about the file name besides the suffix.

2. If one wants to load in auxiliary files to the ``msh`` object, then a cell-array with those filenames is passed to the ``msh ``constructor like...
```
m = msh('fname','ECGC.14','aux',{'myfort.13','myotherfort.15','heyafort.24'}
```

To summarize, we can now add in support for other triangulation/model formats using this design. Further, ``msh`` objects can be institated through just a points and triangulation table matrix.

Commit:
determine file type from suffix

Commit:
typo

Commit:
more bugs, typo fix, and support for opt. called nob

krober10nd commented:
As a reminder: should be a warning like "hey your bandwidth is big, you should probably renumber the mesh. You can renumber by calling ``msh.renum``)

WPringle commented:
I think the changes look good. But we need a `help msh` for it now.

Commit:
adding docstring to mesh class

krober10nd commented:
Ok, added a docstring to msh's class constructor. Also, if there's some bad call, it prints out the usage instructions.

WPringle committed:
Update writefort14.m

- adding ability to handle the external weir bcs in writefort.14

WPringle committed:
Update msh.m

Fixing bugs. The fort.14 only and the points and elements thing wasn't working correctly. Also inconsistent use of 'file' and 'fname' in aux loop. Changed all 'file's to 'fname'.

WPringle commented:
Found some bugs that fixed in the latest commit. The fort.14 only and the points and elements thing wasn't working correctly with latest fixes for the set of files. Also inconsistent use of 'file' and 'fname' in aux loop. Changed all 'file's to 'fname'.

WPringle committed:
Update msh.m

fixing the help for 'nob'

WPringle committed:
Update msh.m

adding some help in error message for f24 in case it is listed before f15 in aux cell array

WPringle reviewed (approved):
Support the changes. 
Maybe you want to double-check that all the use cases work. I think it should be good now. I tested on a suite of aux files as well as the point/element manual input and the single fort.14 input.

krober10nd commented:
lgtm.
